### PR TITLE
Added initial group update tests

### PR DESCRIPTION
**Pull request by TheHollidayInn**
Here are my update tests from the previous PR. One test is pending for my invite PR to be merged. I can just force push this PR once the invite routes are merged.

### CODE REVIEW

File: test/api/v3/integration/groups/PUT-groups.test.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+describe('Put /group', () => {
+  let groupLeader;
+  let groupName = 'Test Public Guild';
+  let groupType = 'guild';
+
+  beforeEach(async () => {
+    groupLeader = await generateUser({balance: 1});
+  });
+
+  xit('returns an error when a non group leader tries to update', async () => {
```
Comment (crookedneighbor) [inline test/api/v3/integration/groups/PUT-groups.test.js]: Is this pended because the functionality is not in place yet?

Comment (TheHollidayInn) [inline test/api/v3/integration/groups/PUT-groups.test.js]: Yep, I stated that in my comment when opening the PR.

Comment (crookedneighbor) [inline test/api/v3/integration/groups/PUT-groups.test.js]: Sorry, missed the note and dove right into the code. 

Why is this dependent on the invite route? Can't we just update a user to be part of the group?

Comment (TheHollidayInn) [inline test/api/v3/integration/groups/PUT-groups.test.js]: No worries.

And, yes.... But, we have talked about this a fair amount on Github and in slack. You concluded that we need to use the routes to correctly test and to imitate multiple api requests.

**Comment by paglias** [comment]
@TheHollidayInn invites are now merged, you can use the route if you need to

**Comment by TheHollidayInn** [comment]
Thanks, I'll be updating this!

**Comment by TheHollidayInn** [comment]
Alright this is updated.

File: test/api/v3/integration/groups/PUT-groups.test.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import {
+  generateUser,
+  translate as t,
+} from '../../../../helpers/api-integration.helper';
+
+describe('Put /group', () => {
```
Comment (crookedneighbor) [inline test/api/v3/integration/groups/PUT-groups.test.js]: Nitpick, but our style is to do `PUT /group`

File: test/api/v3/integration/groups/PUT-groups.test.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  let groupLeader;
+  let groupName = 'Test Public Guild';
+  let groupType = 'guild';
+
+  beforeEach(async () => {
+    groupLeader = await generateUser({balance: 1});
+  });
+
+  it('returns an error when a non group leader tries to update', async () => {
+    let groupToUpdate = await groupLeader.post('/groups', {
```
Comment (crookedneighbor) [inline test/api/v3/integration/groups/PUT-groups.test.js]: Let's extract this group creation into the beforeEach

File: test/api/v3/integration/groups/PUT-groups.test.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    let memberToAttemptUpdate = await generateUser();
+
+    await groupLeader.post(`/groups/${groupToUpdate._id}/invite`, {
+      uuids: [memberToAttemptUpdate._id],
+    });
+    await memberToAttemptUpdate.post(`/groups/${groupToUpdate._id}/join`);
+
+    await expect(memberToAttemptUpdate.put(`/groups/${groupToUpdate._id}`, {
+      name: groupUpdatedName,
+    }))
```
Comment (crookedneighbor) [inline test/api/v3/integration/groups/PUT-groups.test.js]: I prefer to put the `.to.eventually.be` stuff on the same line here.

Comment (TheHollidayInn) [inline test/api/v3/integration/groups/PUT-groups.test.js]: Like this

```
await expect(memberToAttemptUpdate.put(`/groups/${groupToUpdate._id}`, {
      name: groupUpdatedName,
    })).to.eventually.be.rejected.and.eql({
      code: 401,
      error: 'NotAuthorized',
      message: t('messageGroupOnlyLeaderCanUpdate'),
    });
```

File: test/api/v3/integration/groups/PUT-groups.test.js (hunk 2 above)
Comment (TheHollidayInn) [inline test/api/v3/integration/groups/PUT-groups.test.js]: #agree

**Pull request merged**